### PR TITLE
Sentiment Strategy Disclosure

### PR DESCRIPTION
**Pull request by rafmacalaba**
Implements sentiment strategy using disclosures from PSE

### CODE REVIEW

**Comment by rafmacalaba** [comment]
tests already included in `python/tests/test_strategies.py` also added a test notebook using collab [here](https://colab.research.google.com/drive/1XCVe86Q2hmR-i5YdeIa_ggXgvItMNPul?usp=sharing) will add a separate PR for the README on the implementation of this. @enzoampil  @jpdeleon

File: python/fastquant/backtest/backtest.py [hunk 1]
```diff
@@ -405,14 +411,19 @@ class CustomData(bt.feeds.PandasData):
...
-        sorted_combined_df.drop(['channel', 'symbol'], axis=1, inplace=True)
+    if (
+        len(
+            set(["channel" "symbol"]).intersection(
+                sorted_combined_df.columns.values
+            )
+        )
+        == 2
+    ):
+        sorted_combined_df.drop(["channel", "symbol"], axis=1, inplace=True)
```
Comment (jpdeleon) [inline python/fastquant/backtest/backtest.py:422]: we might need to extend the character length in pyproject.toml for black formatter longer than the default

**Review by jpdeleon — APPROVED** [review]
LGTM! We may need to leave a note somewhere that this module works only for philippine stocks so our international users won't get confused.

File: python/fastquant/backtest/backtest.py (hunk 1 above)
Comment (rafmacalaba) [reply python/fastquant/backtest/backtest.py:422]: @jpdeleon i don't know this.

Comment (jpdeleon) [reply python/fastquant/backtest/backtest.py:422]: Ah it is just the formatting option here: https://github.com/enzoampil/fastquant/blob/master/python/pyproject.toml
The default is `line-length = 79` and it's probably better to make this a bit longer so the formatting won't cut the lines above awkwardly. Not crucial though, just aesthetics.

Comment (rafmacalaba) [reply python/fastquant/backtest/backtest.py:422]: ah I see. thanks for that :D

File: python/tests/test_strategies.py [hunk 2]
```diff
@@ -41,6 +43,31 @@ def test_backtest():
                 strategy
             )
             assert cerebro is not None, errmsg
+
+            data_disclosures = get_stock_data(
+                "JFC", "2015-01-01", "2020-09-30", source="phisix"
+            )
+
+            sentiments_disclosures = get_disclosure_sentiment(
```
Comment (enzoampil) [inline python/tests/test_strategies.py:51]: Can we make the date range as short as possible? perhaps 1 month? Let's be mindful that this test will run every time a commit happens, which could be a lot of requests to the website.

I suspect we'll get cases where tests will fail due to the site being down.

**Review by enzoampil — CHANGES_REQUESTED** [review]
Generally LGTM. Just hoping we can make the test data pull as short as possible to avoid having too many requests from tests running due to commits.

File: python/tests/test_strategies.py (hunk 2 above)
Comment (enzoampil) [reply python/tests/test_strategies.py:51]: Actually, maybe better if we forego the pse edge data pull and just run the sentiment pipeline on a saved copy of historical disclosures (similar to the previous sentiment tests i believe).

Comment (rafmacalaba) [reply python/tests/test_strategies.py:51]: got it!

**Comment by enzoampil** [comment]
LGTM :smile:

**Pull request merged**